nexus3/pkg/client/client.go

```diff
@@ -32,6 +34,25 @@ func NewClient(config Config) *Client {
 		config.Timeout = &defaultTimeout
 	}
 
+	var caCertPool *x509.CertPool
+	if *config.RootCAPath != "" {
+		caCert, err := os.ReadFile(*config.RootCAPath)
+		// Backwards because we need to return a client and haven't got a logger
+		if err == nil {
+			caCertPool, err := x509.SystemCertPool()
+			if err != nil {
+				caCertPool = x509.NewCertPool()
+			}
+			caCertPool.AppendCertsFromPEM(caCert)
+		}
+	}
+
+	var cert tls.Certificate
+	if *config.ClientCertificatePath != "" && *config.ClientKeyPath != "" {
```

Comment:
please add nil handling for config.ClientCertificatePath and config.ClientKeyPath , see test pipeline